contracts/gateway/GraphTokenGateway.sol

```diff
@@ -54,4 +57,10 @@ abstract contract GraphTokenGateway is GraphUpgradeable, Pausable, Managed, ITok
     function paused() external view returns (bool) {
         return _paused;
     }
+
+    /**
+     * @dev Runs state validation before unpausing, reverts if
+     * something is not set properly
+     */
+    function _checksBeforeUnpause() internal view virtual;
```

Comment:
How about turning this function into a modifier? 

```
    function setPaused(bool _newPaused) external onlyGovernorOrGuardian onlyValidState(_newPaused) {
        if (!_newPaused) {
            _checksBeforeUnpause();
        }
        _setPaused(_newPaused);
    }
```

Reply:
hmm on second thought it looks kind of weird since the new paused state shouldn't be related to the state validity

Reply:
Yeah, using a modifier could work, but it did feel a bit weird so I think I prefer it as a function... I don't feel strongly about it either way

Reply:
Yeah i think it looks weird so lets keep it as is.